proximal/tests/test_backend.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+
+from proximal.experimental.ir.lin_ops import FFTConv, Grad, MultiplyAdd
+from proximal.experimental.ir.problem import Problem, Variable
+from proximal.experimental.ir.prox_fns import GroupNorm, Nonneg, SumSquares
+
+
+def test_complex_problem() -> None:
+    dims = [512, 512]
+    assert Problem(
+        u=Variable(shape=dims),
+        psi_fns=[
+            SumSquares(
+                lin_ops=[
+                    FFTConv(kernel=np.ones(3)),
+                    MultiplyAdd(
+                        scale=0.4,
+                        offset=-np.ones(dims),
+                    ),
+                ],
+            ),
+            GroupNorm(
+                alpha=1e-5,
+                lin_ops=[
+                    Grad(),
+                ],
+            ),
+            SumSquares(
+                alpha=1e-3,
+                lin_ops=[
+                    Grad(),
+                ],
+            ),
+            Nonneg(
+                lin_ops=[],
+            ),
+        ],
```

Comment:
> This is very cool! I do think there are some downsides to having the user write a string rather than construct the expression tree in native Python code. However, it's a reasonable approach to experiment with.

Yes, problem composition from string may deter new users. To write in native Python, IR backend is available for direct composition (see above), but I personally find it too verbose. 

Perhaps we can implement and support two frontend for evaluations: one composed from string, another composed from native python code. To see which one is more elegant.